### PR TITLE
Fix fx2trt CI test trigger condition

### LINKED ISSUE #70969: fx2trt tests not running?
I'm trying to piece together what is going on with our fx2trt tests. As far as I can tell, https://github.com/pytorch/pytorch/pull/70203 implemented an entirely separate job config with no sharding that runs all our tests, but DOESN'T run fx2trt tests. 

Here is an example recent job from master: https://github.com/pytorch/pytorch/runs/4732473836?check_suite_focus=true. If you search the raw logs, there are no fx2trt tests run at all. This job takes almost 2 hours because it's not sharded and runs a completely redundant set of tests, wasting capacity and time.

Recently, https://github.com/pytorch/pytorch/pull/70658 was landed. I'm not sure what this PR does from the description. It looks like it enables fx2trt tests on our default shards. If we're doing this, why do we have a separate fx2trt job? Can we delete the separate job? Furthermore, fx2trt are not enabled by this PR either, and are not currently running anywhere in master as far as I can tell.

Something seems seriously off; am I missing something?

cc @ezyang @gchanan @zou3519 @bdhirsh @seemethere @malfet @pytorch/pytorch-dev-infra

### PR DESCRIPTION
**Pull request by wushirong**
Summary: Replace test trigger with test_config matching.

Test Plan: CI

Differential Revision: D33480971

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -893,7 +893,7 @@ def get_selected_tests(options):
     # Only run fx2trt test with specified option argument
     if options.fx2trt_tests:
         selected_tests = list(
-            filter(lambda test_name: "fx2trt" in test_name, selected_tests)
+            filter(lambda test_name: test_name in FX2TRT_TESTS, selected_tests)
         )
     else:
         options.exclude.extend(FX2TRT_TESTS)
```
Comment (janeyx99) [inline test/run_test.py:899]: This should be done in a separate PR, but I think it's more clear to not automatically exclude these tests but use the --exclude-fx2trt-tests flag to trigger whether or not to run them.

**Review by janeyx99 — APPROVED** [review]

File: test/run_test.py (hunk 1 above)
Comment (malfet) [reply test/run_test.py:899]: IMO it would be better to skip the tests if PyTorch is not built with TensorRT
